src/Maestro/Maestro.Authentication/AuthenticationConfiguration.cs

```diff
@@ -71,14 +72,26 @@ public static void ConfigureAuthServices(this IServiceCollection services, IConf
         string entraRole = entraAuthConfig["UserRole"]
             ?? throw new Exception("Expected 'UserRole' to be set in the Entra configuration containing " +
                                    "a role on the application granted to API users");
+        string? redirectUri = entraAuthConfig["RedirectUri"];
 
         var openIdAuth = services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme);
 
         openIdAuth
             .AddMicrosoftIdentityWebApi(entraAuthConfig, EntraAuthorizationPolicyName);
 
         openIdAuth
-            .AddMicrosoftIdentityWebApp(entraAuthConfig);
+            .AddMicrosoftIdentityWebApp(options =>
+            {
+                entraAuthConfig.Bind(options);
+                if (!string.IsNullOrEmpty(redirectUri))
+                {
+                    options.Events.OnRedirectToIdentityProvider += context =>
```

Comment:
```suggestion
                    // URI where the BarViz auth loop will come back to.
                    // This is needed because the service might run under a different hostname (like the Container App one),
                    // whereas we need to redirect to the proper domain (e.g. maestro.dot.net)
                    options.Events.OnRedirectToIdentityProvider += context =>
```